Sources/NIO/FileHandle.swift

```diff
@@ -84,11 +84,20 @@ extension NIOFileHandle {
     /// Open a new `NIOFileHandle`.
     ///
     /// - parameters:
-    ///     - path: the path of the file to open. The ownership of the file descriptor is transferred to this `NIOFileHandle` and so it will be closed once `close` is called.
-    public convenience init(path: String) throws {
+    ///     - forReadingAtPath: the path of the file to open for reading. The ownership of the file descriptor is transferred to this `NIOFileHandle` and so it will be closed once `close` is called.
+    public convenience init(forReadingAtPath path: String) throws {
         let fd = try Posix.open(file: path, oFlag: O_RDONLY | O_CLOEXEC)
         self.init(descriptor: fd)
     }
+
+    /// Open a new `NIOFileHandle`.
+    ///
+    /// - parameters:
+    ///     - forReadingAtPath: the path of the file to open for writing. The ownership of the file descriptor is transferred to this `NIOFileHandle` and so it will be closed once `close` is called. If file does not exist, it will be created with permissions 0644.
+    public convenience init(forWritingAtPath path: String) throws {
+        let fd = try Posix.open(file: path, oFlag: O_WRONLY | O_CREAT, mode: S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH)
```

Comment:
Where are all these extra mode flags coming from? Why is `O_CLOEXEC` missing? I think the differences between this and the other initialiser need to be explained or removed.

Reply:
What do you mean? We need to specify mode for `O_CREAT`, otherwise it creates a file with 0000 permissions

Reply:
added O_CLOEXEC

Reply:
Hrm. Can we add those mode flags as an argument to this method and give it that default value?

Reply:
Done